src/star/star_align_reads/script.py

```diff
@@ -0,0 +1,112 @@
+import tempfile
+import subprocess
+import shutil
+from pathlib import Path
+
+## VIASH START
+par = {
+    "input": [
+        "src/star/star_align_reads/test_data/a_R1.1.fastq",
+        "src/star/star_align_reads/test_data/a_R1.2.fastq",
+    ],
+    "input_r2": [
+        "src/star/star_align_reads/test_data/a_R2.1.fastq",
+        "src/star/star_align_reads/test_data/a_R2.2.fastq",
+    ],
+    "genomeDir": "src/star/star_align_reads/test_data/genome.fasta",
+    "aligned_reads": "aligned_reads.sam"
+}
+meta = {
+    "cpus": 8,
+    "temp_dir": "/tmp"
+}
+## VIASH END
+
+##################################################
+# check and process SE / PE R1 input files
+input_r1 = par["input"]
+readFilesIn = ",".join(par["input"])
+par["input"] = None
+
+# check and process PE R2 input files
+input_r2 = par["input_r2"]
+if input_r2 is not None:
+    if len(input_r1) != len(input_r2):
+        raise ValueError("The number of R1 and R2 files do not match.")
+    readFilesIn = [readFilesIn, ",".join(par["input_r2"])]
+    par["input_r2"] = None
+
+# store readFilesIn
+par["readFilesIn"] = readFilesIn
+
+##################################################
+
+# determine readFilesCommand
+if input_r1[0].endswith(".gz"):
+    print(">> Input files are gzipped, setting readFilesCommand to zcat", flush=True)
+    par["readFilesCommand"] = "zcat"
+elif input_r1[0].endswith(".bz2"):
+    print(">> Input files are bzipped, setting readFilesCommand to bzcat", flush=True)
+    par["readFilesCommand"] = "bzcat"
+
+##################################################
+# store output paths
+expected_outputs = {
+    "aligned_reads": ["Aligned.out.sam", "Aligned.out.bam"],
+    "reads_per_gene": "ReadsPerGene.out.tab",
+    "chimeric_junctions": "Chimeric.out.junction",
+    "log": "Log.final.out",
+    "splice_junctions": "SJ.out.tab",
+    "unmapped": "Unmapped.out.mate1",
+    "unmapped_r2": "Unmapped.out.mate2"
+}
+output_paths = {name: par[name] for name in expected_outputs.keys()}
+for name in expected_outputs.keys():
+    par[name] = None
+
```

Comment:
If these keys are not needed anymore:
```suggestion
output_paths = {name: par.pop(name) for name in expected_outputs.keys()}
```

Reply:
This makes it less clear to me what is happening. If I look at a dictionary expression like this, I don't expect it to have side-effects on `par` without really reading the code.

Reply:
Sure, I can understand that